scripts/generate_ego_trajectory.py

```diff
@@ -0,0 +1,84 @@
+#!/usr/bin/python3
+
+import subprocess
+import sys
+import os
+from pathlib import Path
+
+
+def run_mola_lidar_odometry(rosbag_path, output_dir):
+    """Run mola-lidar-odometry-cli with the given ROS bag file and output directory."""
+
+    # Validate input file
+    if not os.path.isfile(rosbag_path):
+        print(
+            f"Error: ROS bag file '{rosbag_path}' does not exist.",
+            file=sys.stderr,
+        )
+        sys.exit(1)
+
+    # Validate output directory
+    if not os.path.isdir(output_dir):
+        print(
+            f"Error: Output directory '{output_dir}' does not exist.",
+            file=sys.stderr,
+        )
+        sys.exit(1)
+
+    # Extract rosbag filename without extension
+    rosbag_name = Path(rosbag_path).stem
+
+    # Define output file paths
+    output_tum_path = os.path.join(output_dir, f"{rosbag_name}_trajectory.tum")
+    output_map_path = os.path.join(output_dir, f"{rosbag_name}_map.simplemap")
+
+    # Use default config file
+    mola_config_path = "$(ros2 pkg prefix mola_lidar_odometry)/share/mola_lidar_odometry/pipelines/lidar3d-default.yaml"
+
+    # Construct the command
```

Comment:
This is a bit mental.
Would it not be easier / more maintainable to setup a bash script to call mola instead of crafting the command in python?